xcode/Subconscious/Shared/Components/Feed.swift

```diff
@@ -12,17 +12,96 @@ import os
+                                if let author = entry.author {
+                                    StoryEntryView(
+                                        story: StoryEntry(
+                                            author: author,
+                                            entry: entry
+                                        ),
+                                        action: { address, _ in
+                                            store.send(.detailStack(.pushDetail(
+                                                MemoDetailDescription.from(
+                                                    address: address,
+                                                    fallback: ""
+                                                )
+                                            )))
+                                        }
+                                    )
+                                }
+                            }
+                            
+                        } else {
+                            StoryPlaceholderView(bioWidthFactor: 1.2)
+                            StoryPlaceholderView(delay: 0.25, nameWidthFactor: 0.7, bioWidthFactor: 0.9)
+                            StoryPlaceholderView(delay: 0.5, nameWidthFactor: 0.7, bioWidthFactor: 0.5)
+                        }
+                    }
+                    .background(Color.secondaryBackground)
+                    
+                    if let count = store.state.entries?.count,
+                       count == 0 {
+                        EmptyStateView()
+                    } else {
+                        FabSpacerView()
+                    }
+
+                }
+                .refreshable {
+                    app.send(.syncAll)
+                }
+                .onAppear {
+                    store.send(.fetchFeed)
+                }
+                .navigationTitle("Feed")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .navigationBarLeading) {
+                        Button(
+                            action: {
+                                store.send(.detailStack(.requestOurProfileDetail))
+                            }
+                        ) {
+                            Image(systemName: "person")
+                        }
+                    }
+                    ToolbarItemGroup(placement: .principal) {
+                        HStack {
+                            Text("Feed").bold()
+                        }
+                    }
+                    ToolbarItem(placement: .primaryAction) {
+                        Button(
+                            action: {
+                                app.send(.presentSettingsSheet(true))
+                            }
+                        ) {
+                            Image(systemName: "gearshape")
+                        }
+                    }
```

Comment:
If we're re-using this, we may be able to factor it out into ToolbarContent https://swiftwithmajid.com/2020/07/15/mastering-toolbars-in-swiftui/#:~:text=iOS%20and%20macOS.-,ToolbarContent,-We%20can%20also

Reply:
I meant the whole toolbar, but YMMV. Non-blocking.